packages/init/src/index.ts

```diff
@@ -27,6 +30,11 @@ export default function initializeInquirer(...args: string[]): Function | void {
         const idx = packages.indexOf(PATH_PREFIX);
         // Retrieve the path supplied along with --generation-path
         generationPath = packages[idx + 1];
+        // Throw an error in case an invalid value is supplied
+        if (generationPath.startsWith('-')) {
+            logger.error('Please specify a valid path');
+            process.exit(2);
+        }
```

Comment:
There are a lot of other edge cases, we really need to switch on arg parser here

Reply:
Indeed, the most likely being supplying other flags just after the `--generation-path` flag, which is what this PR aims to address.